services/app-api/utils/formTemplates/formTemplates.test.ts

```diff
@@ -62,8 +70,19 @@ const generateReportHash = (
 describe("Test getOrCreateFormTemplate WP", () => {
   beforeEach(() => {
     jest.restoreAllMocks();
+    dynamoClientMock.reset();
   });
   it("should create a new form template if none exist", async () => {
+    dynamoClientMock
+      .on(QueryCommand)
+      // mocked once for search by hash
+      .resolvesOnce({
+        Items: [],
+      })
+      // mocked again for search for latest report
+      .resolvesOnce({
+        Items: [],
+      });
```

Comment:
Idea: this mocking library allows targeting calls by argument, so we _could_ make use of that like so:
```suggestion
    dynamoClientMock
      .on(QueryCommand, {
        KeyConditionExpression: "reportType = :reportType AND md5Hash = :md5Hash",
      })
      .resolves({ Items: [] });
    dynamoClientMock
      .on(QueryCommand, {
        KeyConditionExpression: "reportType = :reportType",
        ScanIndexForward: false,
      })
      .resolves({ Items: [] });
```

This is a small win, in that the mocking is more explicit. But it also comes at a small cost, in that the mocking is more verbose. Occasionally jest's `.mockReturnValueOnce` and friends rely on inappropriate knowledge of the order in which the function under test makes its calls... but that's not really the case here, and you could argue that we would only be swapping it for inappropriate knowledge of the calls' arguments.

Stepping back to a bigger picture, it would be nice if the two query methods (`getTemplateVersionByHash` and `getNewestTemplateVersion`) were elsewhere, so that they could be mocked by name here. That would both avoid any inappropriate knowledge and prevent the need to directly mock the dynamo client for this business-logic test. But that is also a bigger refactor than we probably want to incorporate into this PR.

tl;dr: This is fine but not great. You could change it if you wanted, to something else... that would also be just fine and not great. 🤷‍♂️

Reply:
Thanks for calling this out. I think I'll leave it for now, knowing that all of the code surrounding this could stand a refactor.